lms/djangoapps/experiments/utils.py

```diff
@@ -60,6 +61,9 @@ def get_experiment_user_metadata_context(course, user):
 
     upgrade_link, upgrade_date = check_and_get_upgrade_link_and_date(user, enrollment, course)
     has_staff_access = has_staff_access_to_preview_mode(user, course)
+    forum_roles = []
+    if user.is_authenticated:
+        forum_roles = list({role.name for role in Role.objects.filter(users=user, course_id=course.id)})
```

Comment:
Nit: you could use `filter(...).values_list('name').distinct()` to do all this work on the database side.